### PR TITLE
Use rust version of svdtools

### PR DESCRIPTION
**Pull request by ithinuel**
Required updates in rp2040-hal are available as draft at: https://github.com/rp-rs/rp-hal/pull/656

### CODE REVIEW

File: update.sh [hunk 1]
```diff
@@ -5,13 +5,18 @@ set -ex
 cargo install --version 0.21.0 svd2rust
 cargo install --version 0.8.0  form
 rustup component add rustfmt
-pip3 install --upgrade --user "svdtools==0.1.23"
+rm -r .venv
+python3 -mvenv .venv
```
Comment (jannic) [inline update.sh:9]: This could be written as `python3 -mvenv --clear` if I understand the docs at https://docs.python.org/3/library/venv.html correctly.

**Comment by jannic** [comment]
I don't know much about venv, but I read the docs at https://docs.python.org/3/library/venv.html, and as I understand it, the suggested change makes a lot of sense.

File: update.sh (hunk 1 above)
Comment (ithinuel) [reply update.sh:9]: Oh, indeed, I learned something more today :)

Thinking about the addition in #61.
Do we want to keep using the python version or simply migrate to the new rust version?

Comment (jannic) [reply update.sh:9]: If both work equally well, I'd prefer the rust version. But do they?

@burrbull,  @adamgreig, what would you recommend?

Comment (adamgreig) [reply update.sh]: No reason not to use the Rust one. If anything it's more maintained.

File: update.sh [hunk 2]
```diff
@@ -1,25 +1,17 @@
...
-
 cargo install --version 0.27.0 svd2rust
 cargo install --version 0.10.0  form
 rustup component add rustfmt
-if [ "$SVDTOOLS" == "svdtools" ]; then
-    cargo install --version 0.2.6 svdtools
-else
-    pip3 install --upgrade --user "svdtools==0.1.25"
-fi
+cargo install --version 0.2.6 svdtools
```
Comment (jannic) [inline update.sh:8]: I'd prefer to keep the option of running the python version as an alternative choice. Would it be possible to just change the default?

**Review by jannic — APPROVED** [review]

**Pull request merged**